fv3core/stencils/fvtp2d.py

```diff
@@ -55,24 +59,36 @@ def q_j_stencil(
         q_j = (q * area + fx1 - fx1[1, 0, 0]) / area_with_x_flux
 
 
-@gtscript.function
-def transport_flux(f, f2, mf):
-    return 0.5 * (f + f2) * mf
+def final_fluxes(
+    q_advected_y_x_advected_mean: FloatField,
+    q_x_advected_mean: FloatField,
+    q_advected_x_y_advected_mean: FloatField,
+    q_y_advected_mean: FloatField,
+    x_unit_flux: FloatField,
+    y_unit_flux: FloatField,
+    x_flux: FloatField,
+    y_flux: FloatField,
+):
+    """
+    Compute final x and y fluxes of q from different numerical representations.
 
+    Corresponds roughly to eq. 4.17 of FV3 documentation, except that the flux
+    is in units of q rather than in units of q per interface area per time.
+    This corresponds to eq 4.17 with both sides multiplied by
+    e.g. x_unit_flux / u^* (similarly for y/v).
 
-def transport_flux_xy(
-    fx: FloatField,
-    fx2: FloatField,
-    fy: FloatField,
-    fy2: FloatField,
-    mfx: FloatField,
-    mfy: FloatField,
-):
+    Combining the advection operators in this way is done to cancel leading-order
+    numerical splitting error.
+    """
     with computation(PARALLEL), interval(...):
         with horizontal(region[:, :-1]):
-            fx = transport_flux(fx, fx2, mfx)
+            x_flux = (
+                0.5 * (q_advected_y_x_advected_mean + q_x_advected_mean) * x_unit_flux
+            )
         with horizontal(region[:-1, :]):
-            fy = transport_flux(fy, fy2, mfy)
+            y_flux = (
+                0.5 * (q_advected_x_y_advected_mean + q_y_advected_mean) * y_unit_flux
+            )
```

Comment:
this is a case where it makes sense to use a function (though albeit the name was probably not right). but you can leave as-is if this is more clear, it's not a huge difference either way

Reply:
I'm good with this being a function if we can think of an explanatory name, but I don't think there is one. If I had to pick one I'd say "average y value instantaneously and after the x flux, and then multiply that average value by the unit weighting being advected to give the total amount of the scalar which is y-fluxed", but that's a bit of a mouthful.